third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -913,7 +871,9 @@ where
         )
     }
 
-    /// Call a native functions.
+    /// Call a native functions. If native function is a dispatchable native (i.e., it dynamically
+    /// dispatches to ome target via reflection), returns true. Otherwise, returns false. For
+    /// tracing, it is responsibility of the caller to ensure the outcome is logged in the trace.
     fn call_native<RTTCheck: RuntimeTypeCheck, RTRCheck: RuntimeRefCheck>(
```

Comment:
why not pass the tracing here as well and do it instead of delegating to the caller? we do have the NoOp one, and otherwise the caller logic seems the same?

Reply:
This is because of closures. We can have a call closure to dispatchable native, basically a dynamic dispatch into a dynamic dispatch, which means that we need to do these hops first, and then record both at the same time as currently we need to do 1) instruction succeeds, 2) record closure, 3) record dispatchable native, so at this point native call should return (and you cannot do (2) yet because if it does not, the call closure does not succeed)